BoneCentral/Brain/Utilities/compassTools.js

```diff
@@ -0,0 +1,51 @@
+var vector = require('./vector.js');
+
+var getTrueCompass = function(compv, accelv) {
+    var av      = unitScale(accelv);
+    var right   = unitScale(vector.cross(av, compv));
+    var res     = unitScale(vector.cross(right, av));
+    return res;
+};
+
+var cdeg = function(compv, va) {
```

Comment:
Figuring out the compensation part of the degree function was the hardest part to write! What I am doing there is realigning the measured accelerometer vector with the Z-axis and then constraining it (somehow) to the X-axis.

The compass vector is transformed with respect to the acceleration vector (ie if the accelerometer tilts forward, so does the compass). Ideally this all seems to work, but introduce noise...

NOTE: I tried online methods of compass compensation prior to this development, but all were severely lacking what we needed or just didn't work.